master/internal/stream/system_intg_test.go

```diff
@@ -1020,3 +1018,367 @@ func TestProjectUpdate(t *testing.T) {
 
 	runUpdateTest(t, pgDB, testCases)
 }
+
+func TestUpdatesOutOfSpec(t *testing.T) {
+	pgDB, dbCleanup := db.MustResolveNewPostgresDatabase(t)
+	t.Cleanup(dbCleanup)
+
+	uid := 1
+	newExperiment3 := streamdata.Experiment{
+		ID:                   3,
+		JobID:                "test_job2",
+		ModelDefinitionBytes: []byte{},
+		OwnerID:              (*model.UserID)(&uid),
+		State:                model.CanceledState,
+		ProjectID:            2,
+	}
+
+	testCases := []updateTestCase{
+		{
+			startupCase: startupTestCase{
+				description: "startup test case for: update experiments where one is not in subscription",
+				startupMsg: buildStartupMsg("1", map[string]string{ExperimentsUpsertKey: "1"},
+					map[string]map[string][]int{ExperimentsUpsertKey: {ExperimentsUpsertKey: {1}}}),
+				expectedSync:      "key: sync_msg, sync_id: 1",
+				expectedUpserts:   []string{},
+				expectedDeletions: []string{"key: experiments_deleted, deleted: "},
+			},
+			description: "update experiments where one is not in subscription",
+			queries: []streamdata.ExecutableQuery{
+				streamdata.GetUpdateExperimentQuery(streamdata.Experiment{
+					ID:    2,
+					State: model.CanceledState,
+				}),
+				streamdata.GetUpdateExperimentQuery(streamdata.Experiment{
+					ID:    1,
+					State: model.CanceledState,
+				}),
+			},
+			expectedUpserts: []string{
+				// potential race condition where the test won't pick up a test failure.
```

Comment:
question: To resolve this could we monitor the publishLoop, so that we see when the event comes in & confirm that it's not sent?